synapse/push/mailer.py

```diff
@@ -123,6 +124,84 @@ def __init__(self, hs, app_name, template_html, template_text):
 
         logger.info("Created Mailer for app_name %s" % app_name)
 
+    @defer.inlineCallbacks
+    def send_threepid_validation(
```

Comment:
this doesn't really feel like it belongs in `Mailer`. In my head, `Mailer` is only concerned about formatting and sending emails; the session management stuff belongs elsewhere. I think `IdentityHandler` might be a better place.